aws/openid/github/openid.tf

```
@@ -0,0 +1,74 @@
+data "aws_caller_identity" "current" {}
+data "aws_region" "current" {}
+
+resource "aws_iam_role" "github_oidc_role" {
+  name = "github-action-${var.name}"
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [{
+      Action = "sts:AssumeRoleWithWebIdentity",
+      Effect = "Allow",
+      Principal = {
+        Federated = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:oidc-provider/token.actions.githubusercontent.com"
+      },
+      "Condition" : {
+        "StringEquals" : {
+          "token.actions.githubusercontent.com:aud" : "sts.amazonaws.com"
+        }
+        "StringLike" : {
+          "token.actions.githubusercontent.com:sub" : "repo:${var.repo}:*"
+        },
+      }
+    }]
+  })
+
+  inline_policy {
+    name = "ReadWriteS3StateBucket"
+    policy = jsonencode({
+      Version : "2012-10-17",
+      Statement : [
+        {
+          Effect : "Allow",
+          Action : [
+            "s3:GetObject",
+            "s3:ListBucket",
+            "s3:PutObject"
+          ],
+          Resource : [
+            "arn:aws:s3:::${var.name}-tfstate",
+            "arn:aws:s3:::${var.name}-tfstate/**"
+          ]
+        }
+      ]
+    })
+  }
+
+  inline_policy {
+    name = "DynamoDbStateLock"
+    policy = jsonencode({
+      Version : "2012-10-17",
+      Statement : [
+        {
+          Effect : "Allow",
+          Action : [
+            "dynamodb:GetItem",
+            "dynamodb:ListItem",
```

Based on the documentation it seems to be DescribeTable instead of listitem 🤔 [src](https://developer.hashicorp.com/terraform/language/settings/backends/s3#dynamodb-table-permissions)

```suggestion
            "dynamodb:DescribeTable",
```